tests/integration/integration_nodejs_test.go

```diff
@@ -2954,10 +2954,16 @@ func TestNodejsComponentProviderRun(t *testing.T) {
 				PrepareProject: func(info *engine.Projinfo) error {
 					providerPath := filepath.Join(info.Root, "..", "provider")
 					installNodejsProviderDependencies(t, providerPath)
-					cmd := exec.Command("pulumi", "package", "add", providerPath)
+
+					cmd := exec.Command("pulumi", "install")
```

Comment:
The tests worked without installing dependencies for Python and YAML. For Python because pulumi is a dependency of the generated provider, and all dependencies are in a flat namespace. For yaml because yaml.

However for pnpm this doesn't work, because pnpm will (arguably correctly) hide indirect dependencies, so `@pulumi/pulumi` is in `node_modules/the-component-provider/node_modules/@pulumi/pulumi`.